src/Symfony/Component/Form/Tests/Extension/Validator/EventListener/ValidationListenerTest.php

```diff
@@ -66,7 +66,7 @@ protected function setUp()
 
     private function getConstraintViolation($code = null)
     {
-        return new ConstraintViolation($this->message, $this->messageTemplate, $this->params, null, 'prop.path', null, null, $code, new Form());
+        return $this->getMock('Symfony\Component\Validator\ConstraintViolation', array('getConstraint'), array($this->message, $this->messageTemplate, $this->params, null, 'prop.path', null, null, $code));
```

Comment:
this was my first step. Than I realized that it is probably a good idea to test, if `getConstraint` is called by `validateForm`. Otherwise it could be simply removed without breaking the tests (which would also fix the bug). So I changed it to build a mock and added `$violation->expects($this->atLeastOnce())` to the test cases.